### PR TITLE
#79

### PR DESCRIPTION
**Pull request by ultimatefiend**
#79

### CODE REVIEW

File: innovation.game.php [hunk 1]
```diff
@@ -7300,6 +7300,22 @@ function stPlayerInvolvedTurn() {
...
                 break;
+				
+            // id 117, Artifacts age 1: Electrum Stater of Efesos
+            case "117N1":
+                while(true) {
+                    $card = self::executeDraw($player_id, 3, 'revealed'); // "Draw and reveal a 1"
+                    $top_card = self::getTopCardOnBoard($player_id, $card['color']);
+                    if ($top_card == null) 
+					{ // "If you do not have a top card of the drawn card's color"
+                        self::transferCardFromTo($card, $player_id, 'board', false, true); // "meld it"
```
Comment (micahstairs) [inline innovation.game.php:7311]: Please drop the last two parameters:

self::transferCardFromTo($card, $player_id, 'board');

Setting the last parameter to true actually makes this transfer count as a "score" towards the monument achievement.

File: innovation.game.php [hunk 2]
```diff
@@ -7300,6 +7300,22 @@ function stPlayerInvolvedTurn() {
                 self::setGameStateValue('card_id_3', self::executeDraw($player_id, $top_green_card_age, 'hand')['id']);
                 $step_max = 2; // --> 2 interactions: see B
                 break;
+				
+            // id 117, Artifacts age 1: Electrum Stater of Efesos
+            case "117N1":
+                while(true) {
+                    $card = self::executeDraw($player_id, 3, 'revealed'); // "Draw and reveal a 1"
```
Comment (micahstairs) [inline innovation.game.php:7307]: The comment should read: "Draw and reveal a 3"

File: innovation.game.php [hunk 3]
```diff
@@ -7300,6 +7300,22 @@ function stPlayerInvolvedTurn() {
...
+                    $top_card = self::getTopCardOnBoard($player_id, $card['color']);
+                    if ($top_card == null) 
+					{ // "If you do not have a top card of the drawn card's color"
+                        self::transferCardFromTo($card, $player_id, 'board', false, true); // "meld it"
+                        continue; // "Repeat this effect"
+                    }
+                    self::notifyGeneralInfo(clienttranslate('There was a top card of matching color.'));
```
Comment (micahstairs) [inline innovation.game.php:7314]: This effect seems straightforward enough that we could do away with this `notifyGeneralInfo` call. These calls are particularly useful when it's hard to trace the logic of what is happening.

**Review by micahstairs — CHANGES_REQUESTED** [review]
Looking good! Just a few minor changes to make.

File: innovation.game.php (hunk 1 above)
Comment (ultimatefiend) [reply innovation.game.php:7311]: done

File: innovation.game.php (hunk 2 above)
Comment (ultimatefiend) [reply innovation.game.php:7307]: done

File: innovation.game.php (hunk 3 above)
Comment (ultimatefiend) [reply innovation.game.php:7314]: done

**Review by micahstairs — APPROVED** [review]
Thanks!

**Pull request merged**